### ISSUE TITLE
A matrix update during resetup process

### ISSUE THREAD

**Issue by lengfeih** [NVIDIA/AMGX#127]
Hi,

I notice that if I update the matrix entries value and keep the matrix sparsity pattern the same, I may able to use the resetup function to reuse the previous step coarsening structure. The structure_reuse_levels option needs to be set to "-1".

My question is although the structure is reused, computeAOperator function will still re-compute the sparsity before fill the Ac matrix, is it possible to skip this sparsity recomputing? 

Thanks
Frank

**Comment by mattmartineau** [comment]
I completely agree that if resetup is redundantly re-computing the sparsity pattern then we could optimise here. I am planning to take a look through the setup phase over the coming months to isolate improvements of this manner: (a) aggressively caching and (b) improving the scaling of the resetup. 

Once I have validated this is what is happening and created a patch I will associate this issue.
